XiaoMi/pegasus
# feat(rocksdb): Support to open rocksdb with or without meta CF

### What problem does this PR solve? 
[Pegasus 2.0](https://github.com/XiaoMi/pegasus/issues/536) has released many new features, as well as many code refactors and bugfixes since the previous release version [1.12.3](https://github.com/XiaoMi/pegasus/releases/tag/v1.12.3). And more important, 2.0 will introduce some uncompatible features, such as update rocksdb from 5.9.2 to 6.6.4, update 'pegasus data version' from 0 to 1, etc.
This patch provide an effort for Pegasus 1.x to be forward compatible with 2.0. When upgrade Pegasus server from 1.x to 2.0 and find some bugs in 2.0, you can downgrade Pegasus to this version.

**NOTE**: 
- Pegasus version (let's suppose it is 1.12.4) with this patch is forward compatible with 2.0.0
- Pegasus 1.12.3 and lower versions are not forward compatible with this patch!!!

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
1. bootstrap a new cluster, with version 1.12.3
2. write and verify data set A
3. upgrade to version 2.0
4. write data set B, verify data set A and B
5. downgrade to version 1.12.4 (the version include this patch)
6. write data set C, verify data set A, B and C
7. upgrade to version 2.0
8. write data set D, verify data set A, B, C and D

Side effects

- Possible performance regression
No
- Increased code complexity
No
- Breaking backward compatibility
No

Related changes

- Need to cherry-pick to the release branch
No
- Need to update the documentation
Yes
- Need to be included in the release note
Yes
